tuyaha/devices/base.py

```diff
@@ -36,15 +45,80 @@ def available(self):
     def iconurl(self):
         return self.icon
 
-    def update(self):
+    def _update_data(self, key, value, force_val=False):
+        if self.data:
+            if not force_val and self.data.get(key) is None:
```

Comment:
What is this for?

Reply:
States change are stored in device data (cache) to have states updated without having to wait discovery or query method to report it back.

Reply:
Sorry, my question was unclear. Why do we need this particular check? Why not to always update cached value? Please, add a comment for this line.

Reply:
I added a comment about this check. Anyway this it is not really important, is a flag to force a behavior that was not original implemented (write a value in device "data" that don't come from API). I put the flag just to not create a possible uncontrolled breaking change, but could also be removed.